docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,216 @@
+
+![A pivot table with a Farm column and Type and Classification rows.](../images/excel-pivots-table-rows-and-columns.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+
+	pivotTable.columnHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+
+	await context.sync();
+});
+```
+
+You can also have a pivot table with only rows or columns.
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+	
+	await context.sync();
+});
+```
+
+## Add data hierarchies to the pivot table
+
+Data hierarchies fill the pivot table with information to combine based on the rows and columns. 
+Adding the data hierarchies of **Crates Sold at Farm** and **Crates Sold Wholesale** gives sums of those figures for each row and column. 
+In the example, both **Farm** and **Type** are rows, with the crate sales as the data. 
+
+![A pivot table showing the total sales of different fruit based on the farm they came from.](../images/excel-pivots-data-hierarchy.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold at Farm"));
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold Wholesale"));
+
+	await context.sync();
+});
+```
+
+## Change aggregation function
+
+Data hierarchies have their values aggregated into a sum by default. The `summarizeBy` parameter defines this behavior based on an `AggregrationFunction` type. 
+
+## Pivot table layouts
+
+A pivot table layout defines the placement of hierarchies and their data. You access the layout to determine the ranges where data is stored. 
+The following code demonstrates how to get the last row from a pivot table by going through the layout.
+
+```ts
+    await Excel.run(async (context) => {
+        let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+        let range = pivotTable.layout.getDataBodyRange();
+        let grandTotalRange = range.getLastRow();
+        grandTotalRange.load();
```

Comment:
Is something missing here? You load the grand total range, but you don't read it after the sync. There doesn't seem to be a point in loading it.